### PR TITLE
feat(registry-access): implement team command in both TypeScript and Rust stacks

### PR DESCRIPTION
**Pull request by kairosci**
## Summary

Implements the `team` command in both the TypeScript CLI and the Rust pacquet port, replacing the previous `not implemented` stub. The command manages organization teams and team memberships on the npm registry, supporting `create`, `destroy`, `add`, `rm`, and `ls` subcommands with `--otp`, `--parseable`, and `--json` flags matching the npm team behavior. On the TypeScript side the new module lives in `pnpm11/registry-access/commands/src/team.ts` and is exported through the existing `@pnpm/registry-access.commands` package. On the Rust side the command follows the same pattern as `dist_tag.rs` with a `TeamArgs` struct, a `TeamError` diagnostic enum, and dispatch through the existing query pipeline.

The PR also implements the npm team API in **pnpr**: `GET /-/org/{scope}/team` and `GET /-/team/{scope}/{team}/user` (plus the `/~<name>/`-prefixed forms) serve listings from the registry's config-declared `teams:` map, gated by the registry-level `access` with denials masked as not-found. Team mutations answer an explicit 403, because pnpr teams are declared in the registry configuration and cannot be changed over HTTP. Both CLI stacks were verified end-to-end against a live pnpr instance.

That end-to-end run surfaced a latent CLI parsing bug, fixed here in `@pnpm/cli.parse-cli-args`: the option parser stopped at the first occurrence of an escape word (`create`, `exec`, `test`) anywhere in argv, so `pnpm team create @org:x --registry <url>` silently dropped the `--registry` override. Escape words now only escape when they are the command itself.

## Squash Commit Body

```text
The team command communicates with the registry through the standard npm team API endpoints.  The scope:team format is parsed to separate the organization scope from the team name.  For mutation subcommands (create, destroy, add, rm) the registry URL is resolved per scope from the registries map with an optional --registry override, the auth header is resolved from the configured credentials honoring scoped credentials, and the request is sent with retry support and bounded response reads.  When an OTP is in play, the Rust side restricts redirects to the configured registry origins so the npm-otp header cannot leak to another host; the TypeScript fetch layer already strips it on cross-host redirects.  The ls subcommand dispatches to listing teams within an org when given @scope and to listing members of a specific team when given @scope:team.  Output supports three modes, the default human-readable listing, --parseable which emits newline-delimited names, and --json which emits structured arrays.

On the TypeScript side the command is registered in pnpm/src/cmd/index.ts and removed from the notImplemented list.  On the Rust side it is added as a CliCommand variant, routed in dispatch, and dispatched in dispatch_query.  Both sides include comprehensive tests covering all subcommands, error paths for 401 403 404 and 409 responses, empty results, and the three output formats.

pnpr serves the npm team API from each hosted registry's config-declared teams: GET /-/org/{scope}/team and GET /-/team/{scope}/{team}/user list teams and members, gated by the registry-level access with denials masked as not-found, while team mutations answer an explicit 403 since pnpr teams are config-managed.

@pnpm/cli.parse-cli-args no longer stops option parsing at an escape word (create, exec, test) that appears as another command's parameter, which previously made pnpm team create drop a trailing --registry option.
```

## Checklist

- The change is implemented in both the TypeScript CLI and the Rust pacquet port
- Added changesets for the affected published packages: @pnpm/registry-access.commands, @pnpm/cli.parse-cli-args, and pnpm
- Added or updated tests

---
Updated by an agent (Claude Code, claude-fable-5): pnpr team API, parse-cli-args escape fix, and review-feedback commits.

### CODE REVIEW

File: pnpm11/registry-access/commands/src/team.ts [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+
+  if (!team) {
+    throw new PnpmError('TEAM_CREATE_NAME_REQUIRED',
+      'Team name is required (e.g., pnpm team create @org:newteam)')
+  }
+
+  const registryUrl = opts.registries?.default ?? 'https://registry.npmjs.org/'
+  const authHeader = getAuthHeaderForRegistry(opts.configByUri, registryUrl)
+  const fetchFromRegistry = createFetchFromRegistry(opts)
+  const otp = opts.cliOptions?.otp
```
[the remark this answers is not in the document]
Comment (kairosci) [reply pnpm11/registry-access/commands/src/team.ts]: fixed in 6c4ab3e827

File: pnpm/crates/cli/src/cli_args/team.rs [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+
+    if response.status().is_success() {
+        return Ok(format!("-{username} removed from @{}:{team}", st.scope));
+    }
+    write_error_from_response(
+        response,
+        format!(r#"remove user "{username}" from team "{}:{team}""#, st.scope),
+    )
+    .await
+}
```
[the remark this answers is not in the document]
Comment (kairosci) [reply pnpm/crates/cli/src/cli_args/team.rs:423]: fixed in 6c4ab3e827

File: pnpm/crates/cli/src/cli_args/team.rs [hunk 3]
```diff
@@ -0,0 +1,602 @@
...
+    }
+    if status == reqwest::StatusCode::CONFLICT {
+        return Err(TeamError::Conflict { body }.into());
+    }
+    Err(TeamError::RegistryWriteFailed { action, status: status.as_u16(), status_text, body }
+        .into())
+}
+
+#[cfg(test)]
+mod tests;
```
[the remark this answers is not in the document]
Comment (kairosci) [reply pnpm/crates/cli/src/cli_args/team.rs:708]: fixed in 6c4ab3e827

**Review by kairosci — COMMENTED** [review]
fixed in 119d944a71

File: pnpm/crates/cli/src/cli_args/team/tests.rs [hunk 4]
```diff
@@ -0,0 +1,157 @@
+use super::{
+    TeamError, TeamInfo, UserInfo, normalize_registry_url, org_team_url, parse_scope_team,
+    render_members, render_teams, team_members_url, team_user_url,
+};
+
+#[test]
+fn parse_scope_team_returns_scope_only() {
+    let st = parse_scope_team("@myorg").expect("should parse");
+    assert_eq!(st.scope, "myorg");
+    assert!(st.team.is_none());
```
[the remark this answers is not in the document]
Comment (kairosci) [reply pnpm/crates/cli/src/cli_args/team/tests.rs:10]: fixed in 7884b62eb8

File: pnpm/crates/cli/src/cli_args/team.rs [hunk 5]
```diff
@@ -0,0 +1,797 @@
...
+        }
+        .into());
+    }
+
+    response
+        .json::<Vec<TeamInfo>>()
+        .await
+        .into_diagnostic()
+        .map_err(|source| registry_operation_error("parsing teams response", source))
+}
```
[the remark this answers is not in the document]
Comment (zkochan) [reply pnpm/crates/cli/src/cli_args/team.rs]: Fixed as of 3878f78eee: responses are now read through a streaming `read_limited_body` helper with `TEAM_BODY_LIMIT` (1 MiB) for the `ls` JSON bodies and `TEAM_ERROR_BODY_LIMIT` (64 KiB) for error bodies, with a truncation marker — the same pattern as `dist_tag.rs`.

---
Written by an agent (Claude Code, claude-fable-5).

File: pacquet/crates/cli/src/cli_args/team.rs [hunk 6]
```diff
@@ -0,0 +1,797 @@
...
+            }
+        }
+    }
+
+    fn context(self, config: &Config) -> miette::Result<TeamContext> {
+        let registry_url = self
+            .registry
+            .clone()
+            .map(normalize_registry_url)
+            .unwrap_or_else(|| config.registry.clone());
```
[the remark this answers is not in the document]
Comment (zkochan) [reply pacquet/crates/cli/src/cli_args/team.rs]: Fixed in b9debed487: `context` now takes `&self` (`fn context<'a>(&self, config: &'a Config)`) and clones the owned fields, so `run` can keep using `self.params` after building the context.

---
Written by an agent (Claude Code, claude-fable-5).

File: pacquet/crates/cli/src/cli_args/team.rs [hunk 7]
```diff
@@ -0,0 +1,797 @@
...
+                retries: config.fetch_retries,
+                factor: config.fetch_retry_factor,
+                min_timeout: Duration::from_millis(config.fetch_retry_mintimeout),
+                max_timeout: Duration::from_millis(config.fetch_retry_maxtimeout),
+            },
+            registry_url,
+            otp: self.otp,
+            parseable: self.parseable,
+            json: self.json,
+        })
```
[the remark this answers is not in the document]
Comment (zkochan) [reply pacquet/crates/cli/src/cli_args/team.rs]: Fixed as of 3878f78eee: `TeamContext` now carries a `registries` map built from `Config::resolved_registries()` (with `--registry` overriding the `default` entry), and every operation resolves its base URL via `registry_for_scope`, which calls `pick_registry_for_package` with a synthetic `@{scope}/_` package name.

---
Written by an agent (Claude Code, claude-fable-5).

File: pnpm/crates/cli/src/cli_args/team.rs [hunk 8]
```diff
@@ -0,0 +1,797 @@
...
+    let mut lines = vec![format!("@{scope}:{team} has the following members:")];
+    for m in &members {
+        lines.push(format!("  {}", m.name));
+    }
+    Ok(lines.join("\n"))
+}
+
+fn auth_header_for_registry(context: &TeamContext<'_>, registry_url: &str) -> Option<String> {
+    context.config.auth_headers.for_url(registry_url)
+}
```
[the remark this answers is not in the document]
Comment (zkochan) [reply pnpm/crates/cli/src/cli_args/team.rs]: Fixed as of 3878f78eee: `auth_header_for_registry` now takes the org scope and resolves credentials with `for_url_with_package(&registry_url, Some("@{scope}/_"))`, so `.npmrc` scoped credentials are honored, matching `dist-tag`.

---
Written by an agent (Claude Code, claude-fable-5).

File: pnpm11/registry-access/commands/src/team.ts [hunk 9]
```diff
@@ -0,0 +1,462 @@
...
+  return new URL(`-/team/${encodeURIComponent(scope)}/${encodeURIComponent(team)}/user`, normalizeRegistryUrl(registryUrl)).href
+}
+
+async function throwRegistryError (response: Response, action: string): Promise<never> {
+  const errorBody = await response.text()
+  const safeErrorBody = [...errorBody]
+    .filter(c => c.charCodeAt(0) > 0x1f && c.charCodeAt(0) !== 0x7f)
+    .join('')
+    .slice(0, 500)
+  if (response.status === 401) {
```
[the remark this answers is not in the document]
Comment (zkochan) [reply pnpm11/registry-access/commands/src/team.ts]: Fixed in 128960f384: `throwRegistryError` now reads the error body through the shared `readErrorBody` helper (extracted from `access.ts` into `common.ts`), which caps the streaming read at 64 KiB before decoding — matching the Rust `TEAM_ERROR_BODY_LIMIT` — and then applies the existing control-character filtering and 500-character display truncation.

---
Written by an agent (Claude Code, claude-fable-5).

File: pnpm11/registry-access/commands/src/team.ts [hunk 10]
```diff
@@ -0,0 +1,462 @@
...
+    json?: boolean
+  }
+  configByUri?: Record<string, RegistryConfig>
+  registries?: Registries
+}
+
+export async function handler (
+  opts: TeamOptions,
+  params: string[]
+): Promise<string> {
```
[the remark this answers is not in the document]
Comment (zkochan) [reply pnpm11/registry-access/commands/src/team.ts:99]: This matches the established style of this package: the `handler` signatures in `distTag.ts` and `owner.ts` use the identical multi-line parameter list without a trailing comma, and the repo lint config accepts it. Keeping `team.ts` consistent with its siblings; no change made.

---
Written by an agent (Claude Code, claude-fable-5).

File: pnpm11/registry-access/commands/src/team.ts [hunk 11]
```diff
@@ -0,0 +1,462 @@
...
+    return { scope: inner }
+  }
+  const scope = inner.slice(0, colonIndex)
+  const team = inner.slice(colonIndex + 1)
+  if (!scope || !team) {
+    throw new PnpmError('TEAM_INVALID_SCOPE',
+      `Team spec must start with @scope, got "${spec}". Use @scope or @scope:team format.`)
+  }
+  return { scope, team }
+}
```
[the remark this answers is not in the document]
Comment (zkochan) [reply pnpm11/registry-access/commands/src/team.ts]: Fixed in 128960f384: `parseScopeTeam` and the other helper functions (`getRegistryAndAuthForOrg`, the URL builders, `throwRegistryError`) now live below their callers at the bottom of the file, following the declarations-after-use convention used by `distTag.ts` and `owner.ts`.

---
Written by an agent (Claude Code, claude-fable-5).

File: pnpm/crates/cli/src/cli_args/team.rs [hunk 12]
```diff
@@ -0,0 +1,674 @@
...
+        &config.tls_by_uri,
+        &NetworkSettings {
+            network_concurrency: config.network_concurrency,
+            fetch_timeout: Duration::from_millis(config.fetch_timeout),
+            user_agent: config.user_agent.clone(),
+        },
+    )
+    .into_diagnostic()
+    .wrap_err("create the network client for team command")
+}
```
[the remark this answers is not in the document]
Comment (zkochan) [reply pnpm/crates/cli/src/cli_args/team.rs:618]: Fixed in 128960f384: when `--otp` is given, the team command now builds its client with `ThrottledClient::for_installs_with_guard`, restricting redirect targets to the configured registry origins (scheme, host, port) — the same defense `access.rs` uses — so a registry-controlled redirect cannot receive the `npm-otp` header.

---
Written by an agent (Claude Code, claude-fable-5).

File: pnpm/crates/cli/src/cli_args/team.rs [hunk 13]
```diff
@@ -0,0 +1,700 @@
...
+                .collect();
+            let guard: RedirectGuard = Arc::new(move |target: &reqwest::Url| -> bool {
+                origins.iter().any(|(scheme, host, port)| {
+                    target.scheme() == scheme
+                        && target.host_str() == Some(host.as_str())
+                        && target.port() == *port
+                })
+            });
+            guard
+        });
```
[the remark this answers is not in the document]
Comment (zkochan) [reply pnpm/crates/cli/src/cli_args/team.rs:293]: This divergence is deliberate and is now called out in a comment at the guard (commit 010f405e60). The guard mirrors the established pacquet pattern — the `access` command installs the identical origin-restricted redirect guard when `--otp` is set — so within the Rust stack `team` is consistent. The behavioral difference from pnpm only shows in the failure mode of an operation that fails either way: on a cross-host redirect of an OTP-carrying request, pnpm strips `authorization`/`npm-otp` and proceeds without credentials (failing auth at the target), while pacquet fails loudly at the redirect hop. Matching pnpm exactly is not possible with a reqwest redirect policy (policies cannot strip custom headers per hop); it needs a manual redirect loop in `pacquet-network`, which would also change `access` — a follow-up beyond this PR.

---
Written by an agent (Claude Code, claude-fable-5).

**Review by zkochan — APPROVED** [review]
